iceoryx_posh/test/moduletests/test_mepoo_chunk_header.cpp

```diff
@@ -108,26 +120,43 @@ TEST_F(ChunkHeader_test, FromPayloadFunctionCalledWithNullptrReturnsNullptr)
 
 TEST_F(ChunkHeader_test, usedChunkSizeIsSizeOfChunkHeaderWhenPayloadIsZero)
 {
-    ChunkHeader sut;
+    constexpr uint32_t CHUNK_SIZE{32U};
+    constexpr uint32_t PAYLOAD_SIZE{0U};
+    ChunkHeader sut{CHUNK_SIZE,
+                    PAYLOAD_SIZE,
+                    iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT,
+                    iox::CHUNK_NO_CUSTOM_HEADER_SIZE,
+                    iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT};
+
     sut.chunkSize = 2 * sizeof(ChunkHeader);
     EXPECT_THAT(sut.usedSizeOfChunk(), Eq(sizeof(ChunkHeader)));
 }
 
 TEST_F(ChunkHeader_test, usedChunkSizeIsSizeOfChunkHeaderPlusOneWhenPayloadIsOne)
 {
-    ChunkHeader sut;
-    sut.chunkSize = 2 * sizeof(ChunkHeader);
+    constexpr uint32_t CHUNK_SIZE{32U};
     constexpr uint32_t PAYLOAD_SIZE{1U};
-    sut.payloadSize = PAYLOAD_SIZE;
+    ChunkHeader sut{CHUNK_SIZE,
+                    PAYLOAD_SIZE,
+                    iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT,
+                    iox::CHUNK_NO_CUSTOM_HEADER_SIZE,
+                    iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT};
+
+    sut.chunkSize = 2 * sizeof(ChunkHeader);
     EXPECT_THAT(sut.usedSizeOfChunk(), Eq(sizeof(ChunkHeader) + PAYLOAD_SIZE));
 }
 
 TEST_F(ChunkHeader_test, usedChunkSizeTerminatesWhenPayloadSizeExceedsChunkSize)
 {
-    ChunkHeader sut;
```

Comment:
I think you require tests for the case:
- creating a chunk header with a custom header size where the custom header alignment does not exceed the chunk header alignment (death test)
- `fromPayload` function
    * acquire payload correct with custom chunk header (cch) & cch alignment <= header alignment
    * acquire payload correct with custom chunk header (cch) & cch alignment > header alignment
    * acquire payload when chunkheader end == aligned payload start
    * acquire payload when chunkheader end != aligned payload start (with backoffset)
    * acquire payload when chunkheader alignment < payload alignment
    * acquire payload when chunkheader alignment >= payload alignment
    * acquire payload when chunkheader alignment < payload alignment (with backoffset)
    * acquire payload when chunkheader alignment >= payload alignment (with backoffset)
- `usedSizeOfChunk` I think it requires the same cases as the payload tests
- `payload` requires also the same test cases ... maybe it makes sense to test always all 3 of them in the same test case
- alignment (maybe this can be handled by an alignment newtype?!)
    * what happens when the alignment is not power of 2?
    * what happen when the alignment is zero?

Reply:
yes, they are not yet done

Reply:
Should be done now. The power of 2 alignment checks will be done with the the `cxx::align_t` type in the next PR